File: ECommerceAIMockUp.Domain/Entities/Product.cs
using ECommerceAIMockUp.Domain.Common;
using ECommerceAIMockUp.Domain.ValueObjects;

namespace ECommerceAIMockUp.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public Gender Gender { get; set; }
    public Season Season { get; set; }
    public double Price { get; set; }

    public int CategoryId { get; set; }
    public Category Category { get; set; }

    public ICollection<ProductDetails> ProductDetails { get; set; }
}

File: ECommerceAIMockUp.Frontend/front-end/src/app/interfaces/product.ts
import { Gender } from "../shared/Enums/gender";
import { Season } from "../shared/Enums/season";

export interface Product {
  id: number;
  name: string;
  price: number;
  description: string;
  categoryName: string; 
  stock: number;
  imageUrl?: string;
  color?: string;
  size?: string;
  gender: Gender; 
  season: Season; 
  categoryId?: number;
}

